src/index_runner/event_loop.py

```diff
@@ -0,0 +1,110 @@
+'''
+The event loop for the index runner.
+'''
+
+import json
+import logging
+import requests
+import time
+import traceback
+
+from typing import Callable, Dict, Any
+
+from confluent_kafka import Consumer, KafkaError
+
+from src.utils.config import config
+
+Message = Dict[str, Any]
+
+# TODO TEST unit tests
+
+
+def start_loop(
+        consumer: Consumer,
+        message_handler: Callable[[Message], None],
+        on_success: Callable[[Message], None] = lambda msg: None,
+        on_failure: Callable[[Message, Exception], None] = lambda msg, e: None,
+        on_config_update: Callable[[], None] = lambda: None,
+        logger: logging.Logger = logging.getLogger('IR')):
+    """
+    Run the indexer event loop.
+
+    Attributes:
```

Comment:
These should be "Args" rather than "Attributes". Attributes are for class or module attributes

Reply:
What's stupid is that while I was writing this I was thinking attributes was a bad name for function arguments.

Reply:
@jayrbolton fixed